tools/integration_tests/util/client/storage_client.go

```diff
@@ -97,3 +101,53 @@ func CreateObjectOnGCS(ctx context.Context, client *storage.Client, object, cont
 
 	return nil
 }
+
+func CreateStorageClientWithTimeOut(ctx *context.Context, storageClient **storage.Client, t *testing.T) func() {
+	var err error
+	var cancel context.CancelFunc
+	*ctx, cancel = context.WithTimeout(*ctx, time.Minute*15)
```

Comment:
Let's make this time configurable and pass a smaller value if the storage client is not persisted throughout the test suite.

Reply:
Let's keep it 5 minutes as for machines with less core it can take more time.